design/layer2-bind-interfaces.md

```diff
@@ -0,0 +1,243 @@
+# Configuring announcing interfaces
+
+## Summary
+
+The purpose of the design is to support configuring the listening nodes and interfaces for each IPAddressPool for L2 mode.
+
+Previous discussions about this topic can be found in [#277](https://github.com/metallb/metallb/issues/277).
+
+## Motivation
+
+There are two motivations:
+
+The first one is that VIP's ARP broadcasts on all interfaces cause loss of network connectivity in some complex scenarios.
+The picture below shows one of my actual scenarios. Virtual interfaces veth0 and veth1 bridge on one physical interface eth0.
+The veth0 is belongs to subnet 192.168.1.0/24, the veth1 is belongs to subnet 192.172.1.0/24. When the Client(192.172.1.2)
+requests its LB Service(VIP is 192.172.1.10), it may send packets to veth0(due to the Speaker advertising VIP from all 
+interfaces), but K8s would reply on veth1, which led to asymmetric routing so that Linux drop the packets from the Client.
+The VIPs should listen to the suitable interfaces according to the network environment, other than all interfaces.
+
+[<img src="./images/complex_system.svg" width="800"/>](./images/complex_system.svg)
+
+Secondly, in a system that clearly distinguishes network planes, people hope to publish VIPs on the corresponding network planes 
+according to the LB Service function, rather than indiscriminately announcing VIPs on each network plane. For example, in the 
+following picture, if one LB Service is only requested by the `management Network`, we hope the Speaker only advertises this VIP 
+on eth0. Of course, this is not necessary.
```

Comment:
I'd mention "Failover ARP spam broadcasts on all interfaces. While again, this is not a security problem in any way, it can confuse dumb networking gear with broken firmware, causing loss of connectivity." as per https://github.com/metallb/metallb/issues/277#issuecomment-559598076